chrome/content/test/posting_process.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @fileOverview Test suite for lib/posting_process.js
+ *
+ **/
+/* global describe, it, expect, spyOn, g */
+describe("Posting Process Suite", function() {
+
+  var pp = g.postingProcess;
+
+  pp.errorNotification = {
+    show: function() { return; },
+    hide: function() { return; },
+  };
+
+  pp.pendingNotification = {
+    show: function() { return; },
+    hide: function() { return; },
+  };
+  
+  var worker =  {
+    port: {
+      emit: function(message_id, message) { return; },
+      on: function(message_id, callback) { return; },
+    },
+    destroy: function() { return; },
+  };
+
+  beforeEach(function() {
+    pp.postingApplicationTab = {
+      close: function() { return; },
+    };
+    pp.postingResultTab = {
+      close: function() { return; },
+      activate: function() { return; },
+      attach: function() { return worker; },
+    };
+  });
+
+  it("tests menuSetup", function() {
+    expect(pp.pendingNotification).toBeDefined();
+    expect(pp.errorNotification).toBeDefined();
+  });
+    
+  it("tests notificationScript", function() {
+    expect(pp.notificationScript("foobar")).toMatch(/foobar/);
+    expect(pp.notificationScript("error")).toMatch(/error/);
+  });
+
+  it("tests hideNotification", function() {
+    spyOn(pp.errorNotification, "hide");
+    pp.hideNotification("error");
+    expect(pp.errorNotification.hide).toHaveBeenCalled();
+    spyOn(pp.pendingNotification, "hide");
+    pp.hideNotification("pendingPost");
+    expect(pp.pendingNotification.hide).toHaveBeenCalled();
+  });
+
+  it("tests sendBtnStatus", function() {
+    spyOn(worker.port, "emit");
+    // Default option is false
+    expect(g.ls.getItem("Options:DissableButton")).toBe(false);
+    pp.sendBtnStatus("foobar", worker);
+    expect(worker.port.emit).toHaveBeenCalled();
+    // Switch to another option
+    g.ls.setItem("Options:DissableButton", true);
+    pp.sendBtnStatus("foobar", worker);
+    expect(worker.port.emit).toHaveBeenCalled();
+    expect(worker.port.emit.calls.argsFor(0)).toEqual(["privlyBtnStatus", "unchecked"]);
+    expect(worker.port.emit.calls.argsFor(1)).toEqual(["privlyBtnStatus", "checked"]);
+  });
+
+  it("tests saveSecret", function() {
+    pp.saveSecret("foobar");
+    expect(pp.messageSecret).toBe("foobar");
+  });
+ 
+  it("tests sendInitialContent", function() {
```

Comment:
Generally you want to title tests as what the test is testing for, rather than simply what it is testing. Something like `it("sends initial content to the app", function() {`is more informative.